internal/provider/resource_tfe_registry_module_test.go

```diff
@@ -803,6 +803,27 @@ func TestAccTFERegistryModule_invalidWithRegistryNameAndNoModuleProvider(t *test
 		},
 	})
 }
+
+func TestAccTFERegistryModule_deleteModuleProvider(t *testing.T) {
+	org, _ := setupDefaultOrganization(t)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckTFERegistryModuleDestroy,
```

Comment:
Given that each top-level acceptance test makes use of `resource.Test()` which executes `terraform destroy` as part of the test cleanup, ensuring there are no failures when `Delete()` is called, it feels somewhat redundant to test explicitly for a destroy. `testAccCheckTFERegistryModuleDestroy` is a check that more appropriately belongs in go-tfe integration tests.

Reply:
Thanks @sebasslash and @uturunku1. In that case, let's avoid adding code we don't need, or creating tech debt. This was thoroughly tested in go-tfe, and after the merge I can monitor any incoming requests/bugs related to this refactor. 👍